include/engine/guidance/result.hpp

```diff
@@ -0,0 +1,22 @@
+#ifndef OSRM_ENGINE_GUIDANCE_RESULT_HPP_
+#define OSRM_ENGINE_GUIDANCE_RESULT_HPP_
+
+namespace osrm
+{
+namespace engine
+{
+namespace guidance
+{
+class Waypoint;
+class Route;
+
+struct Result
+{
+    std::vector<Waypoint> waypoints;
+    std::vector<Route> routes;
```

Comment:
Just had a look over this changeset - what you're doing here is not allowed: you're storing incomplete types in stdlib containers. The container have to know the `value_type`'s size though.